enginetest/evaluation.go

```diff
@@ -538,6 +538,14 @@ func checkResults(
 				}
 				widenedExpected[i][j] = actual // ensure it passes equality check later
 			}
+			// used for server engine tests
+			if b, ok := widenedExpected[i][j].(bool); ok {
+				if b {
+					widenedExpected[i][j] = int64(1)
+				} else {
+					widenedExpected[i][j] = int64(0)
+				}
+			}
```

Comment:
It would be good to document a little bit more about why we're doing this and how we could remove this in the future. You mention that it's for the server tests, but someone reading this in a year will probably need more context. 

Ideally, we wouldn't need to massage the results coming back like this, so I think if we're going to add in code like this, we should explain why and ideally add a TODO that explains what we need to do to remove this special case. For this one in particular, it seems like we either have serialization code in the `SQL` functions that translates booleans to a 1 or 0 when we send them over the wire, or... maybe it's the go-mysql-driver that's showing us different results? Either way, would be good to explain where it's coming from and what we'd have to do to clean it up in the future.